src/network/networkd-link.c

```diff
@@ -454,6 +454,18 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
         if (r < 0)
                 return r;
 
+        r = hashmap_ensure_allocated(&manager->links_by_name, &string_hash_ops);
+        if (r < 0)
+                return r;
+
+        r = hashmap_put(manager->links_by_name, link->ifname, link);
+        if (r < 0)
+                return r;
+
```

Comment:
I think these lines should be moved to `link_update_by_name()`.

Reply:
The allocation should he here .